File: src/main/java/org/example/imobsyss2/entities/Imovel.java
package org.example.imobsyss2.entities;


import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Imovel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String endereco;

    private String tipo;

    private double valor;

    private double area;

    private int quartos;

    private int banheiros;

    private String descricao;

    private EnumStatusImovel status;

    private String proprietario;




}

File: front/app/types/imovel.ts
export class Imovel{
    constructor(
        public id: number | null,
        public endereco: string,
        public tipo: string,
        public valor: number,
        public area: number,
        public quartos: number,
        public banheiros: number,
        public descricao: string,
        public status: string,
        public proprietario: string

    ){}
}

export interface ImovelFormProps{
    imovelExistente?:Imovel
}